### PR TITLE
[RFC] Enable -Wconversion on path.c (adoption)

### PR DESCRIPTION
**Pull request by cjoach**
ref #567 @justinmk #2842 can be closed.

### CODE REVIEW

File: src/nvim/path.c [hunk 1]
```diff
@@ -745,11 +743,11 @@ static bool is_unique(char_u *maybe_unique, garray_T *gap, int i)
     if (j == i)
       continue;        /* don't compare it with itself */
 
-    candidate_len = (int)STRLEN(maybe_unique);
-    other_path_len = (int)STRLEN(other_paths[j]);
-    if (other_path_len < candidate_len)
-      continue;        /* it's different when it's shorter */
-
+    candidate_len = STRLEN(maybe_unique);
+    other_path_len = STRLEN(other_paths[j]);
```
Comment (philix) [inline src/nvim/path.c]: You can declare the types here since the variables are only used inside the loop block.

File: src/nvim/path.c [hunk 2]
```diff
@@ -773,7 +771,7 @@ static void expand_path_option(char_u *curdir, garray_T *gap)
   char_u      *path_option = *curbuf->b_p_path == NUL
                              ? p_path : curbuf->b_p_path;
   char_u      *buf;
-  int len;
+  size_t len;
```
Comment (philix) [inline src/nvim/path.c]: `len` can be declared three times in the three blocks where it's used. This reduces the scope and the number of uninitialized variables.

File: src/nvim/path.c [hunk 3]
```diff
@@ -1265,10 +1266,10 @@ static int expand_backtick(
   char_u      *cmd;
   char_u      *buffer;
   int cnt = 0;
-  int i;
+  char_u i;
```
Comment (philix) [inline src/nvim/path.c]: This can be declared in the if block it's used below.

File: src/nvim/path.c [hunk 4]
```diff
@@ -2017,12 +2018,13 @@ int expand_wildcards(int num_pat, char_u **pat, int *num_file, char_u ***file,
  */
 int match_suffix(char_u *fname)
 {
-  int fnamelen, setsuflen;
+  size_t fnamelen;
+  size_t setsuflen;
```
Comment (philix) [inline src/nvim/path.c]: Both can be declared in initialization.

**Comment by cacplate** [comment]
@philix done all declaration moves.

**Pull request merged**